binding/web/test/rhino.test.ts

```diff
@@ -0,0 +1,218 @@
+    if (expectedContext !== undefined) {
+      expect(inference.intent).to.deep.eq(expectedContext.intent);
+      expect(inference.slots).to.deep.eq(expectedContext.slots);
+    } else {
+      expect(inference.isUnderstood).to.be.false;
+    }
+  } catch (e) {
+    expect(e).to.be.undefined;
+  }
+};
+
+describe("Rhino Binding", function () {
+  for (const instance of [Rhino, RhinoWorker]) {
+    const instanceString = (instance === RhinoWorker) ? 'worker' : 'main';
+
+    it(`should be able to init with public path (${instanceString})`, () => {
+      cy.wrap(null).then(async () => {
+        await runInitTest(instance);
+      });
+    });
+
+    it(`should be able to init with base64 (${instanceString})`, () => {
+      cy.wrap(null).then(async () => {
+        await runInitTest(instance, {
+          model: { base64: rhinoParams, forceWrite: true }
+        });
+      });
+    });
+
+    it(`should be able to handle UTF-8 public path (${instanceString})`, () => {
+      cy.wrap(null).then(async () => {
+        await runInitTest(instance, {
+          model: { publicPath: '/test/rhino_params.pv', forceWrite: true, customWritePath: '테스트' }
+        });
+      });
+    });
+
+    it(`should be able to handle invalid public path (${instanceString})`, () => {
+      cy.wrap(null).then(async () => {
+        await runInitTest(instance, {
+          model: { publicPath: 'invalid', forceWrite: true },
+          expectFailure: true
+        });
+      });
+    });
+
+    it(`should be able to handle invalid base64 (${instanceString})`, () => {
+      cy.wrap(null).then(async () => {
+        await runInitTest(instance, {
+          model: { base64: 'invalid', forceWrite: true },
+          expectFailure: true
+        });
+      });
+    });
+
+    it(`should be able to handle invalid access key (${instanceString})`, () => {
+      cy.wrap(null).then(async () => {
+        await runInitTest(instance, {
+          accessKey: 'invalid',
+          expectFailure: true
+        });
+      });
+    });
+
+    it(`should be able to handle invalid sensitivity(${instanceString})`, () => {
+      cy.wrap(null).then(async () => {
+        await runInitTest(instance, {
+          context: { publicPath: '/test/contexts/rhino.rhn', forceWrite: true, sensitivity: -1 },
+          expectFailure: true
+        });
+      });
+    });
+
+    for (const testParam of testData.tests.within_context) {
+      it(`should be able to process within context (${testParam.language}) (${instanceString})`, () => {
+        try {
+          const suffix = (testParam.language === 'en') ? '' : `_${testParam.language}`;
+          cy.getFramesFromFile(`audio_samples/test_within_context${suffix}.wav`).then( async pcm => {
+            await runProcTest(
+              instance,
+              pcm,
+              {
+                context: {
+                  publicPath: `/test/contexts/${encodeURIComponent(testParam.context_name.replace("ā", "a").replace("ō", "o"))}_wasm.rhn`,
```

Comment:
Since we use it in a couple places can we make the `.replace("ā", "a").replace("ō", "o")` a quick helper function to help in case we need to update / remove it in the future?